src/plugins/workspace/public/components/workspace_form/utils.ts

```diff
@@ -104,7 +111,8 @@ export const convertPermissionSettingsToPermissions = (
   permissionItems: WorkspacePermissionSetting[] | undefined
 ) => {
   if (!permissionItems || permissionItems.length === 0) {
-    return undefined;
+    // Workspace object should always have permissions, set it as an empty object here instead of undefined.
+    return EMPTY_PERMISSIONS;
   }
```

Comment:
does it has any impact when permission control is turned off?

Reply:
I supposed it may not as this is a specific object for workspace object.

Reply:
when permission control is off, permission field will not initialized and add to kibana index, i am not sure it will cause any issue when try to save a object with empty permission, we can have a test

Reply:
This means the value of permission field will not be consistent between permission is on/off, right? So it seems updating client wrapper will be a better solution?